cl/search/management/commands/generate_opinion_embeddings.py

```diff
@@ -0,0 +1,182 @@
+            "--auto-resume",
+            action="store_true",
+            default=False,
+            help="Auto resume the command using the last document_id logged in Redis. "
+            "If --start-id is provided, it'll be ignored.",
+        )
+        parser.add_argument(
+            "--count",
+            type=int,
+            help="The number of opinions to process.",
+        )
+        parser.add_argument(
+            "--embedding-queue",
+            type=str,
+            default="batch1",
+            help="Which celery queue to use for embedding.",
+        )
+        parser.add_argument(
+            "--upload-queue",
+            type=str,
+            default="batch1",
+            help="which celery queue to use for uploading to S3.",
+        )
+        parser.add_argument(
+            "--throttle-min-items",
+            type=int,
+            default=5,
+            help="The celery throttle min items.",
+        )
+
+    def send_batch(
+        self,
+        batch: list[int],
+        embedding_queue: str,
+        upload_queue: str,
+        database: str,
+    ) -> None:
+        """Send a batch of items for embedding creation and saving to S3.
+
+        :param batch: A list of Opinion IDs representing the batch to process.
+        :param embedding_queue: The name of the queue to process the embedding task.
+        :param upload_queue: The name of the queue to process the saving task.
+        :param database: The database to be used during processing.
+        :return: None.
+        """
+        self.throttle.maybe_wait()
+        chain(
+            create_opinion_text_embeddings.si(batch, database).set(
+                queue=embedding_queue
+            ),
+            save_embeddings.s().set(queue=upload_queue),
+        ).apply_async()
+
+    def handle(self, *args, **options):
+        embedding_queue = options["embedding_queue"]
+        upload_queue = options["upload_queue"]
+        database = options["database"]
+        batch_size = options["batch_size"]
+        count = options.get("count", None)
+        auto_resume = options["auto_resume"]
+        min_opinion_size = settings.MIN_OPINION_SIZE
+        start_id = options["start_id"]
+        throttle_min_items = options["throttle_min_items"]
+        self.throttle = CeleryThrottle(
+            queue_name=embedding_queue, min_items=throttle_min_items
+        )
+        if auto_resume:
+            start_id = get_last_parent_document_id_processed(
+                compose_redis_key()
+            )
+            self.stdout.write(
+                f"Auto-resume enabled starting embedding from ID: {start_id}."
+            )
+
+        opinions = Opinion.objects.filter(id__gte=start_id)
+        # Limit opinions to retrieve if count was provided.
+        opinions_to_process = (
+            opinions[:count] if count is not None else opinions
+        )
+        opinions_with_best_text = opinions.order_by("pk").with_best_text()
```

Comment:
I wonder....if we're doing `with_best_text()`, should we be deferring all the text fields so that we don't fetch them *except* in the best_text field? Or should the model manager be doing that automatically?

Reply:
Yeah. I confirmed and all the fields are being deferred within `with_best_text()`.